server/raft.go

```diff
@@ -2947,11 +2949,11 @@ func (n *raft) runAsCandidate() {
 			nterm := n.term
 			n.RUnlock()
 
-			if vresp.granted && nterm >= vresp.term {
```

Comment:
Why not >=?

Reply:
We should step down if we detect a higher term than the current candidate is campaigning for, so changing to `==` allows the `else` condition to kick in for that case regardless of granted status.

Reply:
Right now with your change we do not handle that at all AFAIK.

Reply:
I think that's right, we would handle the same term if it's a granted vote only, step down for a higher term and ignore a lower term. That way anything still in flight before a new campaign is properly ignored.

Reply:
Each vote request has different inbox, so I do not believe there is a chance of other stuff reaching us from a prior campaign.

Reply:
You are correct, it is the same. Hmm.. ok. We can revisit that in a later PR.